src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamConformanceTests.cs

```diff
@@ -26,8 +26,8 @@ protected override async Task<StreamPair> CreateWrappedConnectedStreamsAsync(Str
 
             await new[]
             {
-                ssl1.AuthenticateAsClientAsync(cert.GetNameInfo(X509NameType.SimpleName, false), null, GetSslProtocols(), false),
-                ssl2.AuthenticateAsServerAsync(cert, false, GetSslProtocols(), false)
+                ssl1.AuthenticateAsServerAsync(cert, false, GetSslProtocols(), false),
+                ssl2.AuthenticateAsClientAsync(cert.GetNameInfo(X509NameType.SimpleName, false), null, GetSslProtocols(), false)
             }.WhenAllOrAnyFailed().ConfigureAwait(false);
```

Comment:
I suggest being explicit the requirement, e.g. rather than this super subtle dependency on the above ordering, adding a read/write in both directions below, with a comment explaining why it's needed.

Reply:
updated. I had that change as well but at the end I decided to with the stream swap. I agree this makes it more obvious.